shared/api/c_api_processor.cc

```diff
@@ -67,59 +69,38 @@ extError_t ORTX_API_CALL OrtxImagePreProcess(OrtxProcessor* processor, OrtxRawIm
     return status.Code();
   }
 
-  auto result_ptr = std::make_unique<ImageProcessorResult>();
+  auto result_ptr = std::make_unique<TensorResult>();
   status =
       processor_ptr->PreProcess(ort_extensions::span(images_ptr->images.get(), images_ptr->num_images), *result_ptr);
   if (status.IsOk()) {
-    *result = static_cast<OrtxImageProcessorResult*>(result_ptr.release());
+    *result = static_cast<OrtxTensorResult*>(result_ptr.release());
   } else {
     *result = nullptr;
   }
 
   return {};
 }
 
-extError_t ORTX_API_CALL OrtxImageGetTensorResult(OrtxImageProcessorResult* result, size_t index, OrtxTensor** tensor) {
-  if (result == nullptr || tensor == nullptr) {
-    ReturnableStatus::last_error_message_ = "Invalid argument";
-    return kOrtxErrorInvalidArgument;
-  }
-
-  auto result_ptr = static_cast<ImageProcessorResult*>(result);
-  ReturnableStatus status(result_ptr->IsInstanceOf(extObjectKind_t::kOrtxKindImageProcessorResult));
-  if (!status.IsOk()) {
-    return status.Code();
-  }
-
-  if (index >= result_ptr->results.size()) {
-    ReturnableStatus::last_error_message_ = "Index out of range";
-    return kOrtxErrorInvalidArgument;
-  }
-
-  auto tensor_ptr = std::make_unique<OrtxObjectWrapper<ortc::TensorBase>>();
-  tensor_ptr->SetObject(result_ptr->results[index].get());
-  *tensor = static_cast<OrtxTensor*>(tensor_ptr.release());
-  return extError_t();
-}
-
-extError_t ORTX_API_CALL OrtxClearOutputs(OrtxProcessor* processor, OrtxImageProcessorResult* result) {
-  if (processor == nullptr || result == nullptr) {
-    ReturnableStatus::last_error_message_ = "Invalid argument";
-    return kOrtxErrorInvalidArgument;
-  }
-
-  const auto processor_ptr = static_cast<const ImageProcessor*>(processor);
-  ReturnableStatus status(processor_ptr->IsInstanceOf(extObjectKind_t::kOrtxKindProcessor));
-  if (!status.IsOk()) {
-    return status.Code();
-  }
-
-  auto result_ptr = static_cast<ImageProcessorResult*>(result);
-  status = result_ptr->IsInstanceOf(extObjectKind_t::kOrtxKindImageProcessorResult);
-  if (!status.IsOk()) {
-    return status.Code();
-  }
-
-  ImageProcessor::ClearOutputs(result_ptr);
-  return extError_t();
-}
+// extError_t ORTX_API_CALL OrtxImageGetTensorResult(OrtxImageProcessorResult* result, size_t index, OrtxTensor**
```

Comment:
nit: did you want to delete this commented code or do we need this OrtxImageGetTensorResult impl checked in? maybe we can just remove it to clean up the code (can be done in future PR) and if we ever need to look at it for reference we could always check the git blame?

Reply:
cleaned up